watchdog/src/metrics.rs

```diff
@@ -33,18 +36,41 @@ pub fn get_metrics() -> CandidHttpResponse {
     }
 }
 
+/// Converts a flag to a gauge value.
+fn flag_to_gauge(flag: Option<Flag>) -> f64 {
+    match flag {
+        None => NO_VALUE,
+        Some(Flag::Enabled) => 1.0,
+        Some(Flag::Disabled) => 0.0,
+    }
+}
+
 /// Encodes the metrics in the Prometheus format.
 fn encode_metrics(w: &mut MetricsEncoder<Vec<u8>>) -> std::io::Result<()> {
-    const NO_VALUE: f64 = f64::NAN;
-
-    let bitcoin_network = crate::storage::get_config().bitcoin_network;
-    let (mainnet, testnet) = match bitcoin_network {
+    let config = crate::storage::get_config();
+    let (mainnet, testnet) = match config.bitcoin_network {
         BitcoinNetwork::Mainnet => (1.0, 0.0),
         BitcoinNetwork::Testnet => (0.0, 1.0),
     };
     w.gauge_vec("bitcoin_network", "Bitcoin network.")?
         .value(&[("network", "mainnet")], mainnet)?
         .value(&[("network", "testnet")], testnet)?;
+    w.encode_gauge(
+        "blocks_behind_threshold",
+        // blocks_behind_threshold must be converted to negative.
+        -1.0 * config.blocks_behind_threshold as f64,
```

Comment:
I don't understand why this comment. Can you elaborate?

Reply:
There are 2 threshold values for the height_diff: blocks_behind_threshold and blocks_ahead_threshold. One must be always negative, another is always positive, eg. [-3, 5]. But if I use say `i64` type, then it'd be technically possible to provide an invalid config, eg. [3, -5]. To avoid that I decided to always store only values, but add a sign when the value is used.

Maybe a better option would be to add an extra function that would handle the sign only in one place.

Reply:
Switched to a getter function. PTAL.